app/assets/scripts/views/country/country.js

```diff
@@ -0,0 +1,279 @@
+  const id = props.match.params.name;
+  const [
+    { locationFetched, locationFetching, locationError, locations },
+    setLocations,
+  ] = useState(defaultLocations);
+  const [
+    { countryFetched, countryFetching, countryError, country },
+    setCountry,
+  ] = useState(defaultCountry);
+
+  useEffect(() => {
+    const fetchLocations = id => {
+      setLocations(state => ({
+        ...state,
+        locationFetching: true,
+        locationError: null,
+      }));
+      let limit = 10000;
+      fetch(`${config.api}/locations?limit=${limit}&country=${id}`)
+        .then(response => {
+          if (response.status >= 400) {
+            throw new Error('Bad response');
+          }
+          return response.json();
+        })
+        .then(
+          json => {
+            setLocations(state => ({
+              ...state,
+              fetched: true,
+              locationFetching: false,
+              locations: json.results,
+            }));
+          },
+          e => {
+            console.log('e', e);
+            setLocations(state => ({
+              ...state,
+              locationFetched: true,
+              locationFetching: false,
+              locationError: e,
+            }));
+          }
+        );
+    };
+
+    const fetchCountry = id => {
+      setCountry(state => ({
+        ...state,
+        countryFetching: true,
+        countryError: null,
+      }));
+      let limit = 1;
+      fetch(`${config.api}/countries?limit=${limit}&country=${id}`)
+        .then(response => {
+          if (response.status >= 400) {
+            throw new Error('Bad response');
+          }
+          return response.json();
+        })
+        .then(
+          json => {
+            setCountry(state => ({
+              ...state,
+              countryFetched: true,
+              countryFetching: false,
+              country: json.results[0],
+            }));
+          },
+          e => {
+            console.log('e', e);
+            setCountry(state => ({
+              ...state,
+              countryFetched: true,
+              countryFetching: false,
+              countryError: e,
+            }));
+          }
+        );
+    };
+
+    fetchLocations(id);
+    fetchCountry(id);
+
+    return () => {
+      setLocations(defaultLocations);
+      setCountry(defaultCountry);
+    };
+  }, []);
+
+  function onDownloadClick(data) {
+    // e && e.preventDefault();
+    props._openDownloadModal(data);
+  }
+
+  // gets sources from locations and uses set to create an array with no duplicates
+  const sourceList = locations
+    ? Array.from(
+        new Set(locations.map(loc => loc.sources[0]).map(JSON.stringify))
+      ).map(JSON.parse)
+    : null;
+
+  let locationGroups = _(locations).sortBy('city').groupBy('city').value();
+
+  return (
+    <>
+      {(countryError || !country) && <ErrorHeader />}
+      {countryFetching && <LoadingHeader />}
+      {country && locations && (
```

Comment:
should we use some if/else ?/: to ensure error, loading, and actual countries page are mutually exclusive?

Reply:
ah yes, that is better